### PR TITLE
stream receive max bytes

### PR DESCRIPTION
**Pull request by tanner0101**
Adds a `maxBytes: Int` argument to `Stream.receive()`

### CODE REVIEW

**Comment by scottbyrns** [comment]
What about making the receive optional? Not all streams would need to be bound by a maximum number of bytes.

File: Sources/Drain.swift [hunk 1]
```diff
@@ -48,9 +48,16 @@ public final class Drain: DataRepresentable, Stream {
         return true
     }
 
-    public func receive() throws -> Data {
-        close()
-        return buffer
+    public func receive(maxBytes: Int) throws -> Data {
```
Comment (tanner0101) [inline Sources/Drain.swift]: Yeah I think making this optional would be nice actually. What do you think @paulo?

Comment (tanner0101) [inline Sources/Drain.swift]: @scottbyrns looping your convo into a line diff so we don't clog the main convo

File: Sources/Drain.swift [hunk 2]
```diff
@@ -22,7 +22,7 @@ public final class Drain: DataRepresentable, Stream {
 
 
         while !stream.closed {
-            if let chunk = try? stream.receive() {
+            if let chunk = try? stream.receive(max: Int.max) {
                 buffer.bytes += chunk.bytes
```
Comment (paulofaria) [inline Sources/Drain.swift:26]: I'm not sure about this here it should be an arbitrary value. Unless we specify in the docs that `Int.max` means implementation default. But I think this is dangerous. It's safer to define something like 1024.

File: Sources/StreamSequence.swift [hunk 3]
```diff
@@ -10,7 +10,7 @@ public final class StreamSequence: Sequence {
             if self.stream.closed {
                 return nil
             }
-            return try? self.stream.receive()
+            return try? self.stream.receive(max: Int.max)
         }
```
Comment (paulofaria) [inline Sources/StreamSequence.swift:14]: I think we should receive an arbitrary `max` here as well.

File: Sources/Drain.swift (hunk 2 above)
Comment (tanner0101) [inline Sources/Drain.swift:26]: We could extend `Int`.

Maybe something like `Int.maxStreamChunkSize` ? That's a little verbose.

Comment (paulofaria) [inline Sources/Drain.swift:26]: honestly I think if we just use an arbitrary number here, were fine.

Comment (tanner0101) [inline Sources/Drain.swift:26]: Okay, let's do 1024 then.

**Comment by paulofaria** [comment]
LGTM

**Pull request merged**